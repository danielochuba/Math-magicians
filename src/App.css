* {
  font-size: 30px;
  border: none;
}

.calculator-container {
  width: 600px;
  height: 600px;
  display: flex;
  flex-wrap: wrap;
  margin: auto;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
}

.body {
  width: 100%;
  background: gray;
}

.display-board {
  font-size: 40px;
}

.buttons {
  display: grid;
  grid-template-columns: auto auto auto auto;
}

.btn {
  height: 100px;
  border-bottom: 1px solid grey;
  border-right: 1px solid grey;
}

.zero {
  grid-column: 1/3;
  grid-row: 5;
}

.dot {
  grid-column: 3;
  grid-row: 5;
}

.equal {
  grid-column: 4;
  grid-row: 5;
}

.operator {
  background: rgb(221, 136, 80);
}

.nav {
  width: 100%;
  height: 100px;
  background: rgb(146, 146, 146);
  text-align: end;
  font-size: 40px !important;
  display: flex;
  justify-content: flex-end;
}
