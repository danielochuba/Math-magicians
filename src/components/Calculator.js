import React, { useState } from 'react';
import calculate from '../logic/calculate';

function CalculatorLogic() {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (e) => {
    const text = e.target.innerText;
    setState((prevState) => calculate(prevState, text));
  };

  const displayValue = state.next
    ? `${state.total || ''} ${state.operation || ''} ${state.next}`
    : state.total || '0';

  return (
    <section>

      <div>
        <div className="nav">
          <p className="display-board">{displayValue}</p>
        </div>
        <div className="buttons">
          <button type="button" onClick={handleClick} className="btn  function">AC</button>
          <button type="button" onClick={handleClick} className="btn  function">+/-</button>
          <button type="button" onClick={handleClick} className="btn  function">%</button>
          <button type="button" onClick={handleClick} className="btn operator">+</button>
          <button type="button" onClick={handleClick} className="btn number">7</button>
          <button type="button" onClick={handleClick} className="btn number">8</button>
          <button type="button" onClick={handleClick} className="btn number">9</button>
          <button type="button" onClick={handleClick} className="btn operator">-</button>
          <button type="button" onClick={handleClick} className="btn number">4</button>
          <button type="button" onClick={handleClick} className="btn number">5</button>
          <button type="button" onClick={handleClick} className="btn number">6</button>
          <button type="button" onClick={handleClick} className="btn operator">x</button>
          <button type="button" onClick={handleClick} className="btn number">1</button>
          <button type="button" onClick={handleClick} className="btn number">2</button>
          <button type="button" onClick={handleClick} className="btn number">3</button>
          <button type="button" onClick={handleClick} className="btn operator">รท</button>
          <button type="button" onClick={handleClick} className="btn number zero">0</button>
          <button type="button" onClick={handleClick} className="btn number dot">.</button>
          <button type="button" onClick={handleClick} className="btn operator equal">=</button>
        </div>
      </div>
    </section>
  );
}

function Calculator() {
  return (
    <>
      <div>
        <h2 style={{ position: 'absolute', top: '50%', left: '2%' }}> Let&apos;s Do some Maths!</h2>
      </div>
      <div className="calculator-container">
        <div className="body">
          <CalculatorLogic />
        </div>
      </div>
    </>
  );
}
export { Calculator, CalculatorLogic };
