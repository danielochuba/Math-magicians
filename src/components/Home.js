/* eslint-disable max-len */
import styles from '../css/Home.module.css';

const Home = () => (
  <div>
    <header className={styles.title}>Welcome to my Page!</header>
    <article>

      <strong>Welcome to Math Magicians!</strong>

      <p>Explore the world of numbers and wisdom with my powerful web application. Whether you need to crunch some numbers or seek inspiration through thought-provoking quotes, we&lsquo;ve got you covered. Our intuitive interface and captivating design make learning and exploring mathematics a delightful experience.</p>

      <strong>Calculator:</strong>
      <p>Access my advanced calculator that can handle complex calculations with ease. Whether you need to perform basic arithmetic operations or dive into more advanced mathematical concepts, my calculator provides the tools you need. Crunch numbers, solve equations, and unlock the magic of mathematics.</p>

      <strong>Quotes:</strong>
      <p>Discover the wisdom of great minds through my Quotes interface. Immerse ymyself in a collection of inspiring quotes that will ignite ymy passion for learning and motivate you on ymy jmyney. From renowned mathematicians to philosophers, my carefully curated quotes will broaden ymy perspective and fuel ymy intellectual curiosity.</p>

      Navigate through the app effortlessly with my user-friendly interface and intuitive navigation system. The Math Magicians web app is designed to provide a seamless and engaging experience for users of all levels.

      Whether you&lsquo;re a student, a professional, or simply a curious mind, Math Magicians is here to empower you with knowledge and inspire you with wisdom. Get ready to embark on an exciting jmyney of mathematical exploration and enlightenment.

      Start ymy adventure now and let Math Magicians be ymy guide!
    </article>
  </div>
);

export default Home;
