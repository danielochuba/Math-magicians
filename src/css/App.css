* {
  border: none;
}

.calculator-container {
  display: flex;
  margin: auto;
  position: absolute;
  bottom: 7%;
  left: 49%;
  right: 2%;
}

.body {
  width: 100%;
}

.display-board {
  font-size: 1.7rem;
  line-height: 34%;
}

.buttons {
  display: grid;
  grid-template-columns: auto auto auto auto;
}

.btn {
  height: 4.5rem;
  border-bottom: 1px solid grey;
  border-right: 1px solid grey;
}

.zero {
  grid-column: 1/3;
  grid-row: 5;
}

.dot {
  grid-column: 3;
  grid-row: 5;
}

.equal {
  grid-column: 4;
  grid-row: 5;
}

.operator {
  background: rgb(221, 136, 80);
}

.nav {
  height: 65px;
  background: rgb(146, 146, 146);
  text-align: end;
  display: flex;
  justify-content: flex-end;
}
