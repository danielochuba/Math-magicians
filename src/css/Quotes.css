.error {
    color: red;
}

.quote-box {
    height: 150px;
    border: 1px solid darkorange;
    margin: auto;
    margin-top: 12px;
    width: 120%;
    text-align: center;
}

.quote-heading {
    margin: 0;
}

.quote {
    font-size: 15px;
}

.author {
    font-style: italic;
    font-size: 15px;
}