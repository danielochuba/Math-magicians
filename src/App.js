import './App.css';
import {
  NavDisplay, LeftSection, Operators,
} from './components/Calculator';

function App() {
  return (
    <div className="calculator-container">
      <NavDisplay />
      <div className="body">
        <LeftSection />
        <Operators />
      </div>

    </div>
  );
}

export default App;
