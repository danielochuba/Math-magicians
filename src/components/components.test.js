import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Home from './Home';
import Quotes from './quotes';
import { CalculatorLogic } from './Calculator';
import App from '../App';

describe('Components renders correctly', () => {
  test('Home Component renders correctly', () => {
    const container = render(<Home />);
    expect(container).toMatchSnapshot();
  });

  test('Quotes Component renders correctly', () => {
    const container = render(<Quotes />);
    expect(container).toMatchSnapshot();
  });

  test('Calculator Component renders correctly', () => {
    const container = render(<CalculatorLogic />);
    expect(container).toMatchSnapshot();
  });
});

test('renders HomePage component on root path', () => {
  render(
    <MemoryRouter initialEntries={['/']}>
      <App />
    </MemoryRouter>,
  );
});

test('renders Calculator component on Calculator path', () => {
  render(
    <MemoryRouter initialEntries={['/calculator']}>
      <App />
    </MemoryRouter>,
  );
});

test('renders QuoteComponent on Quote path', () => {
  render(
    <MemoryRouter initialEntries={['/quote']}>
      <App />
    </MemoryRouter>,
  );
});
