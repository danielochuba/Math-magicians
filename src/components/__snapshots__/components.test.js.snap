// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components renders correctly Calculator Component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <div>
          <div
            class="nav"
          >
            <p
              class="display-board"
            >
              0
            </p>
          </div>
          <div
            class="buttons"
          >
            <button
              class="btn  function"
              type="button"
            >
              AC
            </button>
            <button
              class="btn  function"
              type="button"
            >
              +/-
            </button>
            <button
              class="btn  function"
              type="button"
            >
              %
            </button>
            <button
              class="btn operator"
              type="button"
            >
              +
            </button>
            <button
              class="btn number"
              type="button"
            >
              7
            </button>
            <button
              class="btn number"
              type="button"
            >
              8
            </button>
            <button
              class="btn number"
              type="button"
            >
              9
            </button>
            <button
              class="btn operator"
              type="button"
            >
              -
            </button>
            <button
              class="btn number"
              type="button"
            >
              4
            </button>
            <button
              class="btn number"
              type="button"
            >
              5
            </button>
            <button
              class="btn number"
              type="button"
            >
              6
            </button>
            <button
              class="btn operator"
              type="button"
            >
              x
            </button>
            <button
              class="btn number"
              type="button"
            >
              1
            </button>
            <button
              class="btn number"
              type="button"
            >
              2
            </button>
            <button
              class="btn number"
              type="button"
            >
              3
            </button>
            <button
              class="btn operator"
              type="button"
            >
              ÷
            </button>
            <button
              class="btn number zero"
              type="button"
            >
              0
            </button>
            <button
              class="btn number dot"
              type="button"
            >
              .
            </button>
            <button
              class="btn operator equal"
              type="button"
            >
              =
            </button>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <div>
        <div
          class="nav"
        >
          <p
            class="display-board"
          >
            0
          </p>
        </div>
        <div
          class="buttons"
        >
          <button
            class="btn  function"
            type="button"
          >
            AC
          </button>
          <button
            class="btn  function"
            type="button"
          >
            +/-
          </button>
          <button
            class="btn  function"
            type="button"
          >
            %
          </button>
          <button
            class="btn operator"
            type="button"
          >
            +
          </button>
          <button
            class="btn number"
            type="button"
          >
            7
          </button>
          <button
            class="btn number"
            type="button"
          >
            8
          </button>
          <button
            class="btn number"
            type="button"
          >
            9
          </button>
          <button
            class="btn operator"
            type="button"
          >
            -
          </button>
          <button
            class="btn number"
            type="button"
          >
            4
          </button>
          <button
            class="btn number"
            type="button"
          >
            5
          </button>
          <button
            class="btn number"
            type="button"
          >
            6
          </button>
          <button
            class="btn operator"
            type="button"
          >
            x
          </button>
          <button
            class="btn number"
            type="button"
          >
            1
          </button>
          <button
            class="btn number"
            type="button"
          >
            2
          </button>
          <button
            class="btn number"
            type="button"
          >
            3
          </button>
          <button
            class="btn operator"
            type="button"
          >
            ÷
          </button>
          <button
            class="btn number zero"
            type="button"
          >
            0
          </button>
          <button
            class="btn number dot"
            type="button"
          >
            .
          </button>
          <button
            class="btn operator equal"
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Components renders correctly Home Component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <header
          class="title"
        >
          Welcome to my Page!
        </header>
        <article>
          <strong>
            Welcome to Math Magicians!
          </strong>
          <p>
            Explore the world of numbers and wisdom with my powerful web application. Whether you need to crunch some numbers or seek inspiration through thought-provoking quotes, we‘ve got you covered. Our intuitive interface and captivating design make learning and exploring mathematics a delightful experience.
          </p>
          <strong>
            Calculator:
          </strong>
          <p>
            Access my advanced calculator that can handle complex calculations with ease. Whether you need to perform basic arithmetic operations or dive into more advanced mathematical concepts, my calculator provides the tools you need. Crunch numbers, solve equations, and unlock the magic of mathematics.
          </p>
          <strong>
            Quotes:
          </strong>
          <p>
            Discover the wisdom of great minds through my Quotes interface. Immerse ymyself in a collection of inspiring quotes that will ignite ymy passion for learning and motivate you on ymy jmyney. From renowned mathematicians to philosophers, my carefully curated quotes will broaden ymy perspective and fuel ymy intellectual curiosity.
          </p>
          Navigate through the app effortlessly with my user-friendly interface and intuitive navigation system. The Math Magicians web app is designed to provide a seamless and engaging experience for users of all levels. Whether you‘re a student, a professional, or simply a curious mind, Math Magicians is here to empower you with knowledge and inspire you with wisdom. Get ready to embark on an exciting jmyney of mathematical exploration and enlightenment. Start ymy adventure now and let Math Magicians be ymy guide!
        </article>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <header
        class="title"
      >
        Welcome to my Page!
      </header>
      <article>
        <strong>
          Welcome to Math Magicians!
        </strong>
        <p>
          Explore the world of numbers and wisdom with my powerful web application. Whether you need to crunch some numbers or seek inspiration through thought-provoking quotes, we‘ve got you covered. Our intuitive interface and captivating design make learning and exploring mathematics a delightful experience.
        </p>
        <strong>
          Calculator:
        </strong>
        <p>
          Access my advanced calculator that can handle complex calculations with ease. Whether you need to perform basic arithmetic operations or dive into more advanced mathematical concepts, my calculator provides the tools you need. Crunch numbers, solve equations, and unlock the magic of mathematics.
        </p>
        <strong>
          Quotes:
        </strong>
        <p>
          Discover the wisdom of great minds through my Quotes interface. Immerse ymyself in a collection of inspiring quotes that will ignite ymy passion for learning and motivate you on ymy jmyney. From renowned mathematicians to philosophers, my carefully curated quotes will broaden ymy perspective and fuel ymy intellectual curiosity.
        </p>
        Navigate through the app effortlessly with my user-friendly interface and intuitive navigation system. The Math Magicians web app is designed to provide a seamless and engaging experience for users of all levels. Whether you‘re a student, a professional, or simply a curious mind, Math Magicians is here to empower you with knowledge and inspire you with wisdom. Get ready to embark on an exciting jmyney of mathematical exploration and enlightenment. Start ymy adventure now and let Math Magicians be ymy guide!
      </article>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Components renders correctly Quotes Component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=" quoteLoad"
      >
        <p
          class="quote"
        >
          Loading...
        </p>
        <div
          class="loader"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=" quoteLoad"
    >
      <p
        class="quote"
      >
        Loading...
      </p>
      <div
        class="loader"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
